staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4131,8 +4134,86 @@ type Event struct {
 	// Type of this event (Normal, Warning), new types could be added in the future
 	// +optional
 	Type string `json:"type,omitempty" protobuf:"bytes,9,opt,name=type"`
+
+	// Time when this Event was first observed.
+	// +optional
+	EventTime metav1.MicroTime `json:"eventTime,omitempty" protobuf:"bytes,10,opt,name=eventTime"`
```

Comment:
now it's getting odd. I didn't expect all new (and party duplicated fields) to show up here in the old external type. This is really ugly as it makes the events object very complex for the user by interleaving basically two versions.

Reply:
We typically have added fields.  It might make sense to collect those fields all together under a struct that contains them and makes it clear these are new fields that may overlap with the old fields.

Reply:
I'm fine with that, though it'll make the conversion function even uglier and field selectors harder to use. I'll update this PR with new API once the discussion on how to name stuff will be concluded.

Reply:
Note that some of the existing fields are reused, so that new uses of the API show some information via the old API, so it's not as simple as old vs new. I dropped a link to the proposal working doc so you can see how we got here.

I do think it would make sense to reorder the fields as in events/types.go, and to update comments here about the new preferred fields. 

I do not think we should make the new fields see less first-class here than the old ones by nesting them or prefixing them with `new` or anything.

We should update the type comment to remove the TODO (why should users care?), indicate that this API will be deprecated once the events Event API reaches v1, and encourage developers and other users to try the new API. We can't deprecate this until then, according to our deprecation policy, but we can suggest that people writing new clients should use the new API.

Reply:
Done.